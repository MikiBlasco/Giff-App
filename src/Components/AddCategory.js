import React, {useState} from 'react';
import PropTypes from 'prop-types';


const AddCategory = (props) => {

   const [inputValue, setInputValue] = useState(''); //ponemos " " por que si dejamos vacío es undefined. Luego el del inputvalue en value es también undefined. Entonces nos sale el Warning the "component changing an uncontrolled input..."

   const handleInputChange = (e) => {
       setInputValue(e.target.value)
   }

   const handleSubmit = (e) => {
       e.preventDefault(); 

       //validation (trim para borrar espacios en blanco de de los lados)
       if ( inputValue.trim().length > 2 ) {

           props.setCategories(cat => [inputValue, ...cat]) 
           setInputValue('') // para que se vacíe el placeHolder al hacer submit. 

         }           
   }
  
    return (

        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>

    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory