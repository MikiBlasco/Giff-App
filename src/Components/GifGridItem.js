import React from 'react'
import ProptTypes from 'prop-types'

export const GifGridItem = ({title, url}) => {

  return (

    <div className='card animate__animated animate__bounceIn'>

    <img src={ url } alt={ title } />
    <p> { title } </p>  

    </div>
  )

}

GifGridItem.proptTypes = {
  title: ProptTypes.string.isRequired,
  url: ProptTypes.string.isRequired
}




 
