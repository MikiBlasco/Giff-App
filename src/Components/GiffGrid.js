import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GiffGrid = ({ category }) => {
    //renombramos con la desestructuración la data para que se llame imagenes.
    const {data: images, loading} = useFetchGifs( category ); //state es producto del customHook


  return (
    <>

            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && <p className='animate__animated animate__flash'>Loading...</p> }

            <div className='card-grid'>


                {
                      images.map((img) => 
                        <GifGridItem 
                          key={ img.id }    
                          { ...img } //así mando cada una de las propiedades del objeto "img" como si fueran indendientes.
                          /> )
                }
            
            </div>

    </>

  )
}
