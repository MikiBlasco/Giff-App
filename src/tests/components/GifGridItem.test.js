import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem }  from '../../Components/GifGridItem';


describe('pruebas del componente <GifGridItem />', () => { 

    const title = 'string1'
    const url = 'string2'
    let wrapper = shallow( <GifGridItem title={title} url={url}/> )

    test('should shoe <GifGridItem /> correctly (matching with snapshot', () => { 

        expect(wrapper).toMatchSnapshot();
     })

     test('debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

     })

     test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );        

     })

     test('debe tener animate__bounceIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className')

        expect(className.includes('animate__bounceIn')).toBe(true);
        //expect(div.hasClass('animate__bounceIn')).toBe(true);
        //expect(div.hasClass('animate__bounceIn')).toBeTruthy();

     })

    


 })