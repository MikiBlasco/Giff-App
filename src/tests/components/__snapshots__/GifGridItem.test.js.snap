// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente <GifGridItem /> should shoe <GifGridItem /> correctly (matching with snapshot 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="string1"
    src="string2"
  />
  <p>
     
    string1
     
  </p>
</div>
`;
